src/solvers/base_solver.hpp

```diff
@@ -22,10 +22,10 @@ class BaseSolver {
   mfem::BlockVector *m_block;
 
   /// Essential BC markers
-  mfem::Array<int> m_ess_bdr;
+  std::shared_ptr<mfem::Array<int> > m_ess_bdr;
```

Comment:
why move to `std::shared_ptr < mfem::Array >`?

Reply:
I was thinking that any user-facing wrappers should use vectors instead of MFEM arrays to reduce the dependence on MFEM. It was pretty easy to use the raw pointers to populate a data structure for use with the internal MFEM functions.

Reply:
Ahh I realized I skipped a step in my response. I want the user to pass in a std::vector instead of an mfem::array. This means I need to construct an mfem::array internally on the fly instead of just passing by reference. I am trying to move away from my old way of just using raw pointers.

Reply:
Moved to copying the std::vector into an mfem::array. The performance hit is acceptable since these will typically be small objects.